metricbeat/module/kibana/stats/data.go

```diff
@@ -83,18 +86,17 @@ func eventMapping(r mb.ReporterV2, content []byte) error {
 	var data map[string]interface{}
 	err := json.Unmarshal(content, &data)
 	if err != nil {
-		r.Error(err)
-		return err
+		return errors.Wrap(err, "failure parsing Kibana Stats API response")
 	}
 
 	dataFields, err := schema.Apply(data)
 	if err != nil {
-		r.Error(err)
+		return errors.Wrap(err, "failure to apply stats schema")
```

Comment:
Ties into the discussion above on what our expecation on error is.

Reply:
Yep, same as https://github.com/elastic/beats/pull/8308/files#r222085637. For non-xpack code paths (like this one here) I will update the PR to report one error per event.